src/policy/mod.rs

```diff
@@ -43,22 +44,78 @@ const ENTAILMENT_MAX_TERMINALS: usize = 20;
 /// Trait describing script representations which can be lifted into
 /// an abstract policy, by discarding information.
 /// After Lifting all policies are converted into `KeyHash(Pk::HasH)` to
-/// maintain the following invariant:
+/// maintain the following invariant(modulo resource limits):
 /// `Lift(Concrete) == Concrete -> Miniscript -> Script -> Miniscript -> Semantic`
+/// Lifting from [miniscript.Miniscript], [descriptor.Descriptor] can fail
+/// if the miniscript contains a timelock combination or if it contains a
+/// branch that exceeds resource limits.
+/// Lifting from Concrete policies can fail if it contains a timelock
+/// combination. It is possible that concrete policy has some branches that
+/// exceed resource limits for any compilation, but cannot detect such
+/// policies while lifting. Note that our compiler would not succeed for any
+/// such policies.
 pub trait Liftable<Pk: MiniscriptKey> {
     /// Convert the object into an abstract policy
     fn lift(&self) -> Result<Semantic<Pk>, Error>;
 }
 
+/// Detailed Error type for Policies
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum LiftError {
+    /// Cannot lift policies that have
+    /// a combination of height and timelocks.
+    HeightTimeLockCombination,
+    /// Duplicate Public Keys
+    BranchExceedResourceLimits,
+}
+
+impl error::Error for LiftError {
+    fn description(&self) -> &str {
+        ""
+    }
+    fn cause(&self) -> Option<&error::Error> {
+        None
+    }
+}
```

Comment:
I just checked, and if you replace this whole block with
```
impl error::Error for LiftError {}
```
it compiles even on 1.29.0.